lang/compiler.go

```diff
@@ -165,8 +166,18 @@ type FluxCompiler struct {
 }
 
 func (c FluxCompiler) Compile(ctx context.Context) (flux.Program, error) {
+	query := c.Query
+
+	if tableflux.Enabled() {
```

Comment:
Can you turn this into a new compiler implementation? `TableFluxCompiler` will help to differentiate this.

Reply:
Similarly, can you put that new compiler in a different package? This change causes the build system to now work differently and won't work with our existing build patterns so it'll break everything downstream that consumes flux.